### PR TITLE
Main page

### PR DESCRIPTION
**Pull request by IlanSirisky**
Implemented Main Page -> All/My Pokemons wrapping this Page with different data.
Managing all state inside main page and not in underlying components - tabs, select, input field.
util function to sort the data based on select value.
Cards View component
style fixes after PP.

### CODE REVIEW

File: src/features/cardView/PokemonCardView/PokemonCardView.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import Modal from "../../../components/Modal/Modal";
+import PokemonModalCard from "../../PokemonModalCard/PokemonModalCard";
+import { transformPokemonDataToAttributes } from "../../../utils/transformData";
+
+interface PokemonCardViewProps {
+  data: IPokemonData[];
+}
+
+const PokemonCardView = ({ data }: PokemonCardViewProps) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
```
Comment (ZivCohenMoveo) [inline src/features/cardView/PokemonCardView/PokemonCardView.tsx]: I think you don't need this state
the modal is open when selectedCard is not null, right?

File: src/pages/FightPage/FightPage.tsx [hunk 2]
```diff
@@ -7,43 +7,72 @@ import {
...
+import {
+  calculateCatchRate,
+  calculateDamage,
+  getRandomPokemon,
+} from "../../utils/fightPageFunctions";
 
-const getRandomPokemon = () => {
-  return pokemonsMockData[Math.floor(Math.random() * pokemonsMockData.length)];
-};
+type PlayerTurn = "player" | "opponent";
```
Comment (ZivCohenMoveo) [inline src/pages/FightPage/FightPage.tsx]: move to types file and change to enum

File: src/pages/FightPage/FightPage.tsx [hunk 3]
```diff
@@ -7,43 +7,72 @@ import {
...
   const [selectedPokemon, setSelectedPokemon] = useState<IPokemonData>(
-    getRandomPokemon()
+    getRandomPokemon(myPokemonsData)
   );
-  const [opponentPokemon, setOpponentPokemon] = useState<IPokemonData>(
-    getRandomPokemon()
+  const [opponentPokemon] = useState<IPokemonData>(
+    getRandomPokemon(pokemonsMockData, selectedPokemon)
+  );
+  const [selectedPokemonHp, setSelectedPokemonHp] = useState(
```
Comment (ZivCohenMoveo) [inline src/pages/FightPage/FightPage.tsx]: need both `selectedPokemonHp` and `selectedPokemon`?
is `selectedPokemonHp` current hp? if yes, change the name of the state

File: src/pages/FightPage/FightPage.tsx [hunk 4]
```diff
@@ -7,43 +7,72 @@ import {
...
+    selectedPokemon.hp
+  );
+  const [opponentPokemonHp, setOpponentPokemonHp] = useState(
```
Comment (ZivCohenMoveo) [inline src/pages/FightPage/FightPage.tsx]: need both `opponentPokemonHp` and `opponentPokemon`?
is `opponentPokemonHp` current hp? if yes, change the name of the state

File: src/pages/MainPage/MainPage.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  style? : CSSProperties
+}
+
+const MainPage = ({ pokemonData, headerText, style }: MainPageProps) => {
+  const [selectedTab, setSelectedTab] = useState<string>(tabsOptions[0].label);
+  const [sortBy, setSortBy] = useState<string>("");
+  const [searchValue, setSearchValue] = useState("");
+  const [filteredData, setFilteredData] = useState(pokemonData);
+
+  useEffect(() => {
```
Comment (ZivCohenMoveo) [inline src/pages/MainPage/MainPage.tsx]: need this `useEffect`?
maybe can do this logic in `handleSearchChange`?

File: src/types/pokemonTypes.ts [hunk 6]
```diff
@@ -38,3 +45,324 @@ export type PokemonType =
   | "Dark"
   | "Steel"
   | "Fairy";
+
+export const typeAdvantages: {
```
Comment (ZivCohenMoveo) [inline src/types/pokemonTypes.ts]: move it to consts file

File: src/utils/sortPokemonData.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import { IPokemonData } from "../types/pokemonTypes";
+
+export const sortData = (data: IPokemonData[], sortBy: string) => {
+  switch (sortBy) {
+    case "az":
```
Comment (ZivCohenMoveo) [inline src/utils/sortPokemonData.ts]: change it to enum

File: src/features/cardView/PokemonCardView/PokemonCardView.tsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+          key={item.id}
+          subheadText={item.id}
+          cardTitle={item.name}
+          image={item.imageSrc}
+          cornerText={`${item.px}px`}
+          topCornerIcon={strength}
+          style={{ cursor: "pointer" }}
+        />
+      ))}
+      {selectedCard && (
```
Comment (ZivCohenMoveo) [inline src/features/cardView/PokemonCardView/PokemonCardView.tsx]: now you don't need this `selectedCard &&`

**Review by YuvalLeviMoveoHLS — APPROVED** [review]

**Pull request merged**